portfolio/src/main/java/com/google/sps/data/CommentPersistHelper.java

```diff
@@ -84,41 +128,60 @@ public void deleteComment(long id) {
         // Remove the comment from persistent storage and the comments list
         comments.remove(i);
         datastore.delete(comment.getKey());
+
+        // Remove the comment's image, if it exists
+        BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+        blobstoreService.delete(comment.getBlobKey());
         break;
       }
     }
   }
 
-  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  /** Returns the given comment, found by its ID. Null if it isn't found */
+  public Comment getCommentById(long id) {
```

Comment:
This would be a good spot to consider Optional<Comment> as well. 

That allows client code to do getCommentById(id).orElse()... which enables composing functions together.